blind_75/boggle_iii.py

```diff
@@ -8,7 +8,21 @@
 # Output:  Following words of dictionary are present
 #          GEEKS
 #          QUIZ
+"""
+Pre Calculation of insertion into a Trie
 
+The time complexity is O(NM) where N is the length of the dictionary 
+and M is the length of the longest word
+
+Space complexity is O(26^N) where N is the number of nodes in 
+the Trie and N is the longest word in the dictionary. 
+I.e. if we only allow alphabetical chars, it's 26! size nodes
+
+Searching using a Trie and the Boggle Board
+Time Complexity is 4^(N^2) because there are 4 directions and N is the length 
```

Comment:
nit: technically, I believe the original boggle board is 8 directions, and it would be `O(8^min(N^2, L)` where L is the longest word, and N is the width of the boggle board.